src/ir/ty.rs

```diff
@@ -101,6 +129,25 @@ pub trait TemplateDeclaration {
                  .collect())
         }
     }
+
+    /// Get only the set of template parameters that this item uses. This is a
+    /// subset of `all_template_params` and does not necessarily contain any of
+    /// `self_template_params`.
+    fn used_template_params(&self, ctx: &BindgenContext) -> Option<Vec<ItemId>>
```

Comment:
The idea is to differentiate between these cases:

```
template <> class IsTemplateWithNoParams {};

class IsNotTemplate {};
```

I don't know if it is worth it however, as we eventually treat them identically.